library/Icingadb/Web/Controller.php

```diff
@@ -312,8 +313,29 @@ public function createViewModeSwitcher()
         $viewModeSwitcher = new ViewModeSwitcher(Url::fromRequest());
         $viewModeSwitcher->setAttribute('id', $this->getRequest()->protectId('view-switcher'));
 
+        $prefs = $this->Auth()->getUser()->getPreferences();
+        $viewMode = $prefs->getValue('icingadb', 'view_mode');
+        if (isset($viewMode)) {
+            $viewModeSwitcher->setDefaultViewMode($viewMode);
+        }
+
         $this->params->shift($viewModeSwitcher->getViewModeParam());
 
+        $viewModeSwitcher->on(ViewModeSwitcher::ON_SUCCESS, function (ViewModeSwitcher $viewModeSwitcher) use ($prefs) {
+            $viewMode = $viewModeSwitcher->getPopulatedValue($viewModeSwitcher->getViewModeParam());
+
+            // Quick patch: Save single preference value to session, no matter, if the view mode changes
```

Comment:
What's a quick patch? If it's WIP, make it a TODO, otherwise reword or remove the comment.